.travis.yml

```diff
@@ -36,16 +61,25 @@ script:
   # Generate the font caches in a single process before starting the
   # multiple processes
   - python -c "from matplotlib import font_manager"
-  - if [[ $BUILD_DOCS == false ]]; then mkdir ../tmp_test_dir; fi
-  - if [[ $BUILD_DOCS == false ]]; then cd ../tmp_test_dir; fi
-  - if [[ $BUILD_DOCS == false ]]; then python ../matplotlib/tests.py -sv --processes=8 --process-timeout=300 $TEST_ARGS; fi
-  - if [[ $BUILD_DOCS == true ]]; then cd doc; python make.py html --small; fi
-  # We don't build the LaTeX docs here, so linkchecker will complain
-  - if [[ $BUILD_DOCS == true ]]; then touch build/html/Matplotlib.pdf; fi
-  - if [[ $BUILD_DOCS == true ]]; then linkchecker build/html/index.html; fi
+  - |
+    if [[ $BUILD_DOCS == false ]]; then
+       mkdir ../tmp_test_dir;
+       cd ../tmp_test_dir;
+       python ../matplotlib/tests.py -sv --processes=8 --process-timeout=300 $TEST_ARGS;
+    else
+        cd doc; python make.py html --small;
+        # We don't build the LaTeX docs here, so linkchecker will complain
+        touch build/html/Matplotlib.pdf;
+        linkchecker build/html/index.html;
+    fi
 
 after_failure:
-  - tar cjf result_images.tar.bz2 result_images
-  - if [[ $TRAVIS_PULL_REQUEST == false ]]; then gem install travis-artifacts; fi
-  - if [[ $TRAVIS_PULL_REQUEST == false ]]; then travis-artifacts upload --path result_images.tar.bz2; fi
-  - if [[ $TRAVIS_PULL_REQUEST != false ]]; then echo "The result images will only be uploaded if they are on the matplotlib/matplotlib repo - this is for security reasons to prevent arbitrary PRs echoing security details." else echo https://s3.amazonaws.com/matplotlib-test-results/artifacts/${TRAVIS_BUILD_NUMBER}/${TRAVIS_JOB_NUMBER}/result_images.tar.bz2; fi
+  - |
+    if [[ $TRAVIS_PULL_REQUEST == false ]]; then
```

Comment:
If it is a PR build the private artifacts just are not available so the upload to AWS will fail, this just skips bundling them up and trying the upload if we know it will fail.

A malicious PR can change this, but it will just waste cycles on travis.